### ISSUE TITLE
[a11y]: Allow user to set `aria-label` on Search component

### ISSUE THREAD

**Issue by allisonishida** [carbon-design-system/carbon#16573]
### Package

@carbon/react

### Package version

^1.48.1

### Automated testing tool and ruleset

N/A

### Description

Users should be allowed to set an `aria-label` value for the Search component, rather than setting it from the `placeholder` field

### Reproduction/example

https://react.carbondesignsystem.com/?path=/docs/components-search--overview

### Steps to reproduce

Do an "Inspect" then view the "Elements", and refer to the `div role="search"` div

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by guidari** [comment]
Hey @allisonishida 
The `aria-label` can be set on the `Search` component as expected. Try to update your @carbon/react to the lastest version.

```
 <Search
    size="lg"
    placeholder="Find your items"
    labelText="Search"
    closeButtonLabelText="Clear search input"
    id="search-1"
    onChange={() => {}}
    onKeyDown={() => {}}
    aria-label="this aria-label is working"
  />
```

![Screenshot 2024-05-28 at 14 04 12](https://github.com/carbon-design-system/carbon/assets/52183462/87f3fd65-1c5f-46eb-871d-e7be75065e94)
